sstables/storage.cc

```diff
@@ -163,20 +201,18 @@ void filesystem_storage::open(sstable& sst) {
 
     // Flushing parent directory to guarantee that temporary TOC file reached
     // the disk.
-    sst.sstable_write_io_check(sync_directory, _dir.native()).get();
+    _dir.sync(sst._write_error_handler).get();
 }
 
 future<> filesystem_storage::seal(const sstable& sst) {
     // SSTable sealing is about renaming temporary TOC file after guaranteeing
     // that each component reached the disk safely.
     co_await remove_temp_dir();
-    auto dir_f = co_await open_checked_directory(sst._write_error_handler, _dir.native());
```

Comment:
so here, it would be nice to do `co_await _dir.open(sst._write_error_handler)`
that would do the open first time around, and return a ready future if it's already opened.

Reply:
@bhalevy 

> so here, it would be nice to do `co_await _dir.open(sst._write_error_handler)`
> that would do the open first time around, and return a ready future if it's already opened.

I don't see what that buys us.

This is for the `seal()` member, from patch#2 ('sstables/storage: sync through "opened_directory" in the "easy" cases').

The pre-patch sequence is: (1.1) _open_checked_, (1.2) _flush_, (1.3) _flush_, (1.4) _close_. The new sequence is: (2.1) _sync_, (2.2) _sync_.

The post-patch step (2.1) _sync_ covers pre-patch steps (1.1) _open_checked_ and (1.2) _flush_ both. Namely:
- if the directory was already open, then the open now simply falls away,
- conversely, if the directory was not yet open, then sync() opens it *and also checks for open failure* (`sync` wraps `open_directory` into `do_io_check`)
- sync unconditionally flushes (and checks for errors)

Post-patch step (2.2) _sync_ maps to pre-patch step (1.3) _flush_.

Pre-patch step (1.4) _close_ is intentionally not covered; the directory is supposed to stay open.

Reply:
ok, this isn't critical.
I won't ibsist